src/ScriptBuilder/Saga/SagaDefinitionReader.cs

```diff
@@ -38,6 +45,192 @@ public static bool TryGetSqlSagaDefinition(TypeDefinition type, out SagaDefiniti
         return true;
     }
 
+    static bool TryInferSagaDefinition(TypeDefinition type, out SagaDefinition definition)
+    {
+        var baseType = type.BaseType as GenericInstanceType;
+        definition = null;
+
+        // Class must directly inherit from NServiceBus.Saga<T> so that no tricks can be pulled from an intermediate class
```

Comment:
should that be written down explicitly in the doco?

Reply:
I will add that to the bullet list of edge cases in doco.